internal/application/scale.go

```diff
@@ -0,0 +1,130 @@
+package application
+
+import (
+	"context"
+	"math"
+	"strconv"
+
+	"github.com/epinio/epinio/helpers/kubernetes"
+	"github.com/epinio/epinio/internal/api/v1/models"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+const (
+	instanceKey = "desired"
+)
+
+// Scaling returns the number of desired instances set by a user for the application
+func Scaling(ctx context.Context, cluster *kubernetes.Cluster, appRef models.AppRef) (int32, error) {
+	scaleSecret, err := scaleLoad(ctx, cluster, appRef)
+	if err != nil {
+		return 0, err
+	}
+
+	result, err := strconv.Atoi(string(scaleSecret.Data[instanceKey])) // nolint:gosec // overflow blocked by guards
+	if err != nil {
+		return 0, err
+	}
+
+	// Reject bad values, and assume single instance - Return err better ? Save back, fix resource ?
+	if result <= 0 || result > math.MaxInt32 {
+		result = 1
+	}
+
+	return int32(result), nil
+}
+
+// ScalingSet sets the desired number of instances for the named application.
+// When the function returns the number is saved.
+func ScalingSet(ctx context.Context, cluster *kubernetes.Cluster, appRef models.AppRef, instances int32) error {
+	return scaleUpdate(ctx, cluster, appRef, func(scaleSecret *v1.Secret) {
+		scaleSecret.Data[instanceKey] = []byte(strconv.Itoa(int(instances)))
+	})
```

Comment:
This is the only consumer of `scaleUpdate` thus the only possible implementation of the `modifyScaling` parameter is this one here. Is there a reason not to collapse these 2 methods in one that does this and only this kind of update? I don't see any other way to use `scaleUpdate` (with a different `modifyScaling`).

Reply:
Heh. I can do the inline (after the rebase is in). I copied the file from environment and did not bother with the inline when the number of users shrunk to one. ... And now I am wondering if this update function could be shared among env, scale, and services. ...

Reply:
That goes into the rabbit hole that the associated load functions can be generalized also.

I am declining the inline request, and will make a ticket to rework this part of the system, i.e. generalize the load/update across these state-holding secrets.

Reply:
Ticket is https://github.com/epinio/epinio/issues/832